pkg/k8sclient/clients.go

```diff
@@ -113,7 +116,9 @@ func NewClients(config ClientsConfig) (*Clients, error) {
 		// properly on our own instead of relying on the manager to provide a
 		// client, which might change in the future.
 		mapper, err := apiutil.NewDynamicRESTMapper(rest.CopyConfig(restConfig))
-		if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			return nil, microerror.Mask(timeoutError)
```

Comment:
This could be a breaking change, if any client code relied on `errors.Is(err, context.DeadlineExceeded)` right? What's the benefit of changing it?

Just asking out of curiosity, I checked PMO and we don't rely on this, so I'm OK with this either way.

Reply:
Good point. I did a search through the GS organization and didn't find anyone handling `DeadlineExceeded` except in `gsctl`. I found in `certs` that we wait for a timeout using `time.After` here https://github.com/giantswarm/certs/blob/05a37d211a36f3e1bdef367279311f06f6753bf8/pkg/certs/searcher.go#L276-L277

In short, I think this would be a silent change for all consumers calling this function and gives us the power to catch an error that we would have otherwise passed upwards with `return microerror.Mask(err)`. I don't mind making it a major bump just to make it clear that there's a possibility for breaking code if you think that's a good idea.